CRM/Contribute/BAO/Contribution.php

```diff
@@ -3549,6 +3564,7 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
       }
     }
     $trxn = CRM_Core_BAO_FinancialTrxn::create($params['trxnParams']);
+    $previousLineItem = CRM_Price_BAO_LineItem::getLineItemsByContributionID($params['contribution']->id);
```

Comment:
$params['contribution']->id and $params['prevContribution']->id hold the same id for contribution.